tests/common.go

```diff
@@ -7725,3 +7777,183 @@ func GetPoolCapacityUsed(poolUUID string) (float64, error) {
+				applicationScaleDownMap[name] = 0
+			}
+			err = Inst().S.ScaleApplication(ctx, applicationScaleDownMap)
+			if err != nil {
+				return err
+			}
+			//waiting for volumes to be detached after scale down
+			for _, v := range appVolumes {
+				t := func() (interface{}, bool, error) {
+					apiVol, err := Inst().V.InspectVolume(v.ID)
+					if err != nil {
+						return "", false, err
+					}
+					if len(apiVol.AttachedOn) == 0 {
+						return "", false, nil
+					}
+					return "", true, fmt.Errorf("volume %s is still attached to %s", v.ID, apiVol.AttachedOn)
+				}
+				_, err = task.DoRetryWithTimeout(t, waitResourceCleanup, 10*time.Second)
+				if err != nil {
+					return err
+				}
+			}
+
+			for _, v := range appVolumes {
+				replicaSets, err := Inst().V.GetReplicaSets(v)
+				if err != nil {
+					return err
+				}
+
+				var checksum string
+				var primaryPool string
+				var primaryNode *node.Node
+				for _, rs := range replicaSets {
+					poolUuids := rs.PoolUuids
+					for _, poolUuid := range poolUuids {
+						nodeDetail, err := GetNodeWithGivenPoolID(poolUuid)
+						if err != nil {
+							return err
+						}
+						poolID, err := GetPoolIDFromPoolUUID(poolUuid)
+						if err != nil {
+							return err
+						}
+
+						inspectVolume, err := Inst().V.InspectVolume(v.ID)
+						if err != nil {
+							return err
+						}
+						log.Infof("Getting md5sum for volume %s on pool %s in node %s", inspectVolume.Id, poolID, nodeDetail.Name)
+						//To-Do update the command if set up is dmthin
+						cmd := fmt.Sprintf("/opt/pwx/bin/runc exec -t portworx md5sum /var/.px/%d/%s/pxdev", poolID, inspectVolume.Id)
+						output, err := Inst().N.RunCommand(*nodeDetail, cmd, node.ConnectionOpts{
+							Timeout:         defaultTimeout,
+							TimeBeforeRetry: defaultRetryInterval,
+							Sudo:            true,
+						})
+						if err != nil {
+							return err
+						}
+						vChecksum := strings.Split(strings.TrimSpace(output), " ")[0]
+						if checksum == "" {
+							checksum = vChecksum
+							primaryPool = poolUuid
+							primaryNode = nodeDetail
+						}
+						if checksum != vChecksum {
+							return fmt.Errorf("md5sum of volume [%s] having [%s] on node %s[pool: %s] is not matching with  checksum [%s] on node %s [pool; %s]", inspectVolume.Id, vChecksum, primaryNode.Name, primaryPool, checksum, nodeDetail.Name, poolUuid)
+						}
+						log.Infof("md5sum of volume [%s] having [%s] on node %s[pool: %s] is matching with  checksum [%s] on node %s [pool; %s]", inspectVolume.Id, vChecksum, primaryNode.Name, primaryPool, checksum, nodeDetail.Name, poolUuid)
+					}
+				}
+			}
+
+			//reverting application scale
+			applicationScaleUpMap := make(map[string]int32, len(ctx.App.SpecList))
+
+			for name, scale := range applicationScaleMap {
+				log.InfoD(fmt.Sprintf("scale up app %s to %d", ctx.App.Key, scale))
+				applicationScaleUpMap[name] = scale
+			}
+			err = Inst().S.ScaleApplication(ctx, applicationScaleUpMap)
```

Comment:
Need a log statment here

Reply:
it already added while creating applicationScaleUpMap